data_structures/tree_234.cpp

```diff
@@ -0,0 +1,1287 @@
+/**
+ * @file
+ * @brief A demo 2-3-4 tree implementation
+ * @details Only implement some basic and complicated operations. Other
+ * operations should be easy to added.
+ * @author [liuhuan](https://github.com/fedom)
+ */
+#include <array>     /// for std::array
+#include <cassert>   /// for assert
+#include <fstream>   /// for std::ofstream
+#include <iostream>  /// for std::cout
+#include <memory>    /// for std::unique_ptr
+#include <queue>     /// for std::queue
+#include <string>    /// for std::to_string
+
+/** @brief 2-3-4 tree node class */
+class Node {
+ public:
+    /**
+     * @brief Node constructor
+     * @param item the first value we insert to the node
+     */
+    explicit Node(int item)
+        : count(1),
+          items({{item, 0, 0}}),
+          children({{nullptr, nullptr, nullptr, nullptr}}) {}
+
+    /**
+     * @brief Get the item count that current saved in the node
+     * @return item count
+     */
+    int GetCount() { return count; }
```

Comment:
It is recommended to use `uint64_t`, rather than `int` (check all the code).

Reply:
Hi Panquesito7, I have 2 questions here:
1. The count is in a range of [0, 3]. Does it need to be a uint64_t since it is a node member and that will add extra unnecessary space consumption?
2. For the item, is it better to be a int64_t since it may be a negative value?

Reply:
> 1. The count is in a range of [0, 3]. Does it need to be a uint64_t since it is a node member and that will add extra unnecessary space consumption?

In this case, you may want to use `uint8_t` or `uint16_t.` Even if it's larger than the expected size, it's better, as there could be an error if we use `int` (or smaller sizes).

> 2. For the item, is it better to be a int64_t since it may be a negative value?

Sounds good. 👍